src/edu/csus/ecs/pc2/ui/JListFrame.java

```diff
@@ -0,0 +1,59 @@
+// Copyright (C) 1989-2023 PC2 Development Team: John Clevenger, Douglas Lane, Samir Ashoo, and Troy Boudreau.
+package edu.csus.ecs.pc2.ui;
+
+import java.awt.BorderLayout;
+import java.awt.Dimension;
+import java.awt.HeadlessException;
+import java.util.Arrays;
+import java.util.List;
+
+import javax.swing.JFrame;
+
+/**
+ * Generic List frame.
+ * 
+ * @see JListPane
+ * 
+ * 
+ * @author Douglas A. Lane <pc2@ecs.csus.edu>
+ */
+public class JListFrame extends JFrame {
+	/**
+	 * 
+	 * 
+	 */
+	private static final long serialVersionUID = -5073426131461967909L;
+
+	private JListPane listPane;
+
+	public JListFrame(String title, Object [] items, int[] selecteditems, ISelectedListsSetter selectedListsSetter) throws HeadlessException {
+		super();
+		setMinimumSize(new Dimension(400, 400));
+		setName("JListFrame");
+		setTitle(title);
+		setPreferredSize(new Dimension(400, 400));
+		
+		listPane = new JListPane(this, items, selecteditems, selectedListsSetter);
+		getContentPane().add(listPane, BorderLayout.CENTER);
+		FrameUtilities.centerFrame(this);
+	}
+	
+	public static void main(String[] args) {
+		Object[] items2 = { "One", "Two", "Three", "Four" };
+
+		JListFrame f = new JListFrame("hi", items2, new int[0], new ISelectedListsSetter() {
+
+			@Override
+			public void setSelectedValuesList(List<Object> selectedValuesList, int[] selectedIndices) {
```

Comment:
I'm not clear what this method definition is for, since I don't see any way for it to be invoked.  Am I missing something?  How would this be invoked?

Reply:
Did you run it and see what it does??   As testing code, I am less inclined to spend time adding comments.

Reply:
Yes, I ran it and saw what it did before I made the comment.  But I still don't understand what the method is for (that is, I don't understand why you've overridden the method and provided a body).  My point was, this override occurs for a `JListFrame` that only exists when the `JListFrame`'s `main()` method is invoked.  When someone invokes the `main()` method (presumably for the purpose of testing the default setup of the class, since it doesn't appear there is any reason any PC2 code would ever invoke it), how would this overridden `setSelectedValuesList()` ever get called?  It's not the same as when some PC2 module constructs a `JListFrame`; the invoking code would have access to the reference to the JListFrame and therefore could call the (internal) `setSelectedValuesList()` method.  I just don't understand how it is intended to be used inside the `main()` method.  Maybe I'm missing something about your intent regarding the method (which is why it would help if there were comments, even though this is "for testing"...)